src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -360,13 +360,22 @@
          (middleware-utils/json-response 200))))
 
 (defn describe-task-file
-  [get-code-content env-name code-id module-name file-data]
+  [get-code-content env-name code-id file-data]
   (let [file (:path file-data)
-        base-name (:name file-data)
+        subpath (:name file-data)
+        basename (fs/base-name subpath)
         size (fs/size file)
         sha256 (ks/file->sha256 (io/file file))
         ;; we trust the file path from Puppet, so extract the subpath from file
-        static-path (re-find (re-pattern (str #"[^/]+/" module-name "/tasks/" base-name)) file)
+        static-path (re-find (re-pattern (str #"[^/]+/[a-z][a-z0-9_]*/(?:tasks|files|lib)/.*" basename)) file)
+        pieces (str/split static-path #"/" 4)
+        ;mount (first (re-find #"(tasks|files|lib)" static-path))
+        module-name (pieces 1)
+        mount (pieces 2)
+        rest (pieces 3)
+        file-content (cond (= mount "files") "modules"
```

Comment:
Oh yup. That makes sense.

Reply:
This could be a case statement instead
```
file-content (case mount "files" "modules" "tasks" "tasks" "lib" "plugins")
```

Reply:
Oh that's so much better.

Reply:
Format it however you like, was easiest to just throw it on one file. Docs at https://clojuredocs.org/clojure.core/case.